mlrun/model_monitoring/db/tsdb/timescaledb/timescaledb_operations.py

```diff
@@ -0,0 +1,736 @@
+                )
+                return
+
+            logger.debug(
+                "Discovered project resources for deletion",
+                project=self.project,
+                tables=len(discovered_tables),
+                views=len(discovered_views),
+                schema=schema_name,
+            )
+
+            drop_statements = []
+
+            # Drop materialized views first (they depend on tables)
+            drop_statements.extend(
+                f"DROP MATERIALIZED VIEW IF EXISTS {schema_name}.{view_name} CASCADE"
+                for view_name in discovered_views
+            )
+            # Drop tables second
+            drop_statements.extend(
+                f"DROP TABLE IF EXISTS {schema_name}.{table_name} CASCADE"
+                for table_name in discovered_tables
+            )
+            # Execute all drops
+            if drop_statements:
+                self._connection.run(statements=drop_statements)
+
+                logger.debug(
+                    "Successfully dropped project resources",
+                    project=self.project,
+                    dropped_objects=len(drop_statements),
+                )
+
+            # Check if schema is empty and drop it if so
+            self._drop_schema_if_empty()
+
+        except Exception as e:
+            logger.warning(
+                "Failed to delete all project resources from TimescaleDB",
+                project=self.project,
+                error=mlrun.errors.err_to_str(e),
+            )
+            raise
+
+        logger.debug(
+            "Successfully deleted all project resources from TimescaleDB",
+            project=self.project,
+        )
+
+    def _drop_schema_if_empty(self) -> None:
+        """Drop the schema if it contains no more tables using parameterized query."""
+        try:
+            schema_name = self.tables[mm_schemas.TimescaleDBTables.PREDICTIONS].schema
+
+            # Check if schema has any tables using parameterized query
+            check_stmt = Statement(
+                """
+                SELECT COUNT(*) as table_count
+                FROM information_schema.tables
+                WHERE table_schema = %s
+                """,
+                (schema_name,),
+            )
+
+            result = self._connection.run(query=check_stmt)
+
+            if result and result.data and result.data[0][0] == 0:
+                # Schema is empty, drop it
+                drop_schema_query = f"DROP SCHEMA IF EXISTS {schema_name} CASCADE"
+                self._connection.run(statements=[drop_schema_query])
+
+                logger.debug(
+                    "Dropped empty schema",
+                    project=self.project,
+                    schema=schema_name,
+                )
+        except Exception as e:
+            logger.warning(
+                "Failed to check/drop empty schema",
+                project=self.project,
+                error=mlrun.errors.err_to_str(e),
+            )
```

Comment:
Do we silence this error on purpose? The caller calls this method in a try-except block, logs and raises the error. That's a bit confusing since actually we already silence any errors here.

Reply:
I believe this is optional cleanup that shouldn't fail the main operation. Will add some clarifications to the code though